cli/Cargo.toml

```diff
@@ -37,6 +37,7 @@ solana-faucet = { path = "../faucet", version = "1.4.0" }
 solana-logger = { path = "../logger", version = "1.4.0" }
 solana-net-utils = { path = "../net-utils", version = "1.4.0" }
 solana-remote-wallet = { path = "../remote-wallet", version = "1.4.0" }
+solana-runtime = { path = "../runtime", version = "1.4.0" }
```

Comment:
Would the feature set make more sense in the sdk?

Reply:
I did consider that, not totally opposed to it.  I didn't have a strong reason for why FeatureSet should generally be exposed in an SDK at this point, so I went for keeping it more narrowly scoped (at the expense of adding this new cli dependency!).  

Only the two too commits are relevant for this PR.  The rest of this PR is just a clone of #12376 (including what you're commenting on here)

Reply:
Yup, I know :-)

Reply:
There, fixed.  This PR now only contains the BPFLoader activation rework

Reply:
In order for us to pass the `FeatureSet` to InvokeContext we could either move `FeatureSet` to the sdk, or `InvokeContext` to runtime.  The latter is looking pretty good: https://github.com/solana-labs/solana/pull/12507